compact.go

```diff
@@ -160,7 +161,19 @@ func (c *compactor) Plan() ([][]string, error) {
 		return [][]string{res}
 	}
 
-	return sliceDirs(c.selectDirs(dms)), nil
+	dirs := sliceDirs(c.selectDirs(dms))
+	if len(dirs) > 1 {
+		return dirs, nil
+	}
+
+	// Compact any blocks that have >5% tombstones.
+	for i := len(dms) - 1; i >= 0; i-- {
+		if dms[i].meta.Stats.NumSeries/dms[i].meta.Stats.NumTombstones <= 20 { // 5%
+			return [][]string{dms[i].dir}, nil
+		}
+	}
```

Comment:
Didn't you have something in there that only made us consider blocks with a relatively long time range? 
I still that still makes sense to have as short ones will be compacted soon enough.